docs/javascript-typescript-interop.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Blazor JavaScript and TypeScript interop
+author: danroth27
+description: Learn how to invoke JavaScript functions from .NET and .NET methods from JavaScript.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 06/10/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/javascript-typescript-interop
+---
+# Blazor JavaScript and TypeScript interop
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+A Blazor app can invoke JavaScript functions from .NET and .NET methods from JavaScript code.
+
+## Invoke JavaScript functions from .NET methods
+
+There are times when Blazor .NET code is required to call a JavaScript function. For example, a JavaScript call can expose a platform-specific capability to the Blazor app.
+
+To call a JavaScript function, register the JavaScript function and call it in the .NET method:
+
+1. Register a JavaScript function by calling `Blazor.registerFunction('functionName', functionImplementation)` in JavaScript. The following example registers two functions that echo a message. The `echo` function is synchronous, while the `echoAsync` version is asynchronous:
+
+    ```javascript
+    Blazor.registerFunction(
+      'echo',
+      function(message){
+        return message;
+      });
+
+    Blazor.registerFunction(
+      'echoAsync',
+      function(message){
+        return Promise.Resolve(message);
+      });
+    ```
+
+1. To invoke a JavaScript function from C#, use the `T Invoke<T>(functionName, args)` (synchronous) or `Task<T> InvokeAsync<T>(functionName, args)` (asynchronous) method from the `RegisteredFunction` class. The following example calls the two preceding JavaScript functions from C# code by indicating their function names and passing `message` arguments:
+
+    ```csharp
```

Comment:
Typically you also put the JS interop call in a static .NET method so that it can be easily called. The Blazor class library template follows this pattern.